quickfixj-core/src/test/java/quickfix/SessionTest.java

```diff
@@ -3045,4 +3045,53 @@ public void testSend_ShouldKeepPossDupFlagAndOrigSendingTime_GivenAllowPosDupCon
         assertTrue(sentMessage.getHeader().isSetField(PossDupFlag.FIELD));
         assertTrue(sentMessage.getHeader().isSetField(OrigSendingTime.FIELD));
     }
+
+    @Test
+    public void testValidateFieldsOutOfRange() throws ConfigError, NoSuchFieldException, IllegalAccessException, InvalidMessage, RejectLogon, UnsupportedMessageType, IncorrectTagValue, FieldNotFound, IncorrectDataFormat, IOException {
+
+        final SessionID sessionID = new SessionID(
+                FixVersions.BEGINSTRING_FIX44, "SENDER", "TARGET");
+        UnitTestApplication application = new UnitTestApplication();
+
+        final MessageStoreFactory mockMessageStoreFactory = mock(MessageStoreFactory.class);
+        final CloseableMessageStore mockMessageStore = mock(CloseableMessageStore.class);
+        when(mockMessageStoreFactory.create(sessionID)).thenReturn(mockMessageStore);
+
+        final LogFactory mockLogFactory = mock(LogFactory.class);
+        final CloseableLog mockLog = mock(CloseableLog.class);
+        when(mockLogFactory.create(sessionID)).thenReturn(mockLog);
+
+        DataDictionaryProvider dataDictionaryProvider = mock(DataDictionaryProvider.class);
+        DataDictionary dataDictionary = new DataDictionary("../quickfixj-messages/quickfixj-messages-fix44/src/main/resources/FIX44.modified.xml");
```

Comment:
This dictionary should be in `src/test/resources` folder of `quickfixj-project` project

Reply:
Also in this case we should be able to avoid specifying its path (even relative). Filename should suffice